### DISCUSSION TITLE
Best practice: How granular should by Notifiers be?

### DISCUSSION THREAD

**Discussion by janpeterka** [excid3/noticed#416] [General]
As I'm implementing notifications, I am not really sure how many Notifiers I should have.
I know there's not a right or wrong way, I'm just interested in how different people approach this.

_For context: My app is basically IS for choirs and music bands; having events, announcements, comments,..., nothing complicated._

So what are my options?

## Have just ApplicationNotifier

I my usecase, everything would pretty much work with only one notifier, as I specify record and message in each case.
Is there some downside to that?

## Have ObjectNotifiers

Another option is to split notifiers by objects - `EventNotifier`, `CommentNotifier`,..
It might make orientation easier, and probably also specify how the notification is shown in UI (like `notification_message` or something).

## Have specific Notifier for every event

I can go "all in" and have `EventCreationNotifier`, `CommentUpdateNotifier`,...
This would allow me to specify message in Notifier. It makes sense to me, but at the same time, it seems bit unnecessary.

So, I just want some discussion - how did you think about this stuff when you implement your notification system.

**Comment by excid3** [comment]
Separate notifiers are most useful when you've got different content for each. For example, you may want to send a different email for a new event vs a new comment.

If they're all about the same, then a single notifier might do the trick.
